File: Java/assignment 1/src/NewtonRaphsonSub.java
// Uppsala University
// Department Of IT
// Programming bridging course Autumn 2013
// Java assignment 1
// Student: Knut Lorenzen 810326-T296

public class NewtonRaphsonSub extends NewtonRaphson {

	@Override
	public Complex f( Complex x ) {
 
		return x.multiply( x ).add( x.multiply( 2 ) ).add( 2 );
	}

	@Override
	public Complex f_prim( Complex x ) {

		return x.multiply( 2 ).add( 2 );
	}

}

File: C++/C++ assignment 1/cpp assignment 1/NewtonRaphsonSub.cpp
// Uppsala University
// Department Of IT
// Programming bridging course Autumn 2013
// C++ assignment 1
// Student: Knut Lorenzen 810326-T296
//
//  NewtonRaphsonSub.cpp
//
//  Created by Knut Lorenzen on 31/10/2013.
//  Copyright (c) 2013 Knut Lorenzen. All rights reserved.
//

#include "NewtonRaphsonSub.h"
#include "Complex.h"

Complex NewtonRaphsonSub::f( Complex x ){
   
    return x*x + x * 2 + 2;
}


Complex NewtonRaphsonSub::f_prim( Complex x ){
    
    return x*2 + 2;
}